internal/api/handlers.go

```diff
@@ -252,6 +257,110 @@ func DeleteCertificate(env *Environment) http.HandlerFunc {
 	}
 }
 
+// GetUserAccounts returns all users from the database
+func GetUserAccounts(env *Environment) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		users, err := env.DB.RetrieveAllUsers()
+		if err != nil {
+			logErrorAndWriteResponse(err.Error(), http.StatusInternalServerError, w)
+			return
+		}
+		apiUsers := make([]certdb.UserAPIModel, 0, len(users))
+		for _, user := range users {
+			apiUsers = append(apiUsers, certdb.UserAPIModel{
+				ID:          user.ID,
+				Username:    user.Username,
+				Permissions: user.Permissions,
+			})
+		}
```

Comment:
ideally you can just modify the tags in the struct to automatically marshall the output correctly and not have to do it manually here.

Reply:
Done